### DISCUSSION TITLE
Add React 18 as a peer dependency.

### DISCUSSION THREAD

**Discussion by mielecmichal** [TanStack/query#3452] [General] [answered]
Could you add a newly released React 18.0 as a peer dependency so that we can avoid npm install warning?

```
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.0" from react-query@3.34.19
npm ERR! node_modules/react-query
npm ERR!   react-query@"^3.34.19" from the root project
```

```
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0 || ^18.0.0"
  },
```

**Comment by TkDodo** [comment, marked as answer]
we could, but the current major version is not compatible with concurrent features of react18, so I'd rather not recommend using v3 with it. If the warning (error?) can be worked around in user land, that would be preferable for v3. v4 will switch to `useSyncExternalStore`, which makes it compatible. We have a PR ready and I'll try to release a beta of v4 soon.

**Reply by Dav3whit3** [reply]
Great to hear man 😍. I found myself in a situation where I cannot resolve react-query with tailwind v3 and I hope I can soon.

Btw, Shout out to all you guys, react-query rocks !

**Reply by TkDodo** [reply]
we've released a version on v3 since, but please read the release notes carefully: 
https://github.com/tannerlinsley/react-query/releases/tag/v3.35.0
